### ISSUE TITLE
[Bug] Misaligned highlight when selecting mixin

### ISSUE THREAD

**Issue by Hiroko103** [beefytech/Beef#1002]
Version: Nightly (04/12/2021)

It happens after a single click on the mixin and after a double click also.

The screenshot shows the code at: corlib/IO/OpenFileDialog.bf:414
![misaligned_highlight](https://user-images.githubusercontent.com/6961586/117037760-201bc800-ad07-11eb-9d86-c89ec5bde2a7.PNG)

**Comment by EinBurgbauer** [comment]
This was a long hunt but it stems from here:
BfExprEvaluator.cpp:15113
```cpp
autoComplete->mInsertStartIdx = targetNameNode->GetSrcStart();
autoComplete->mInsertEndIdx = targetNameNode->GetSrcEnd();
```
Commenting this out seems to fix it. If that code snippet runs, mInsert.. is already set correctly, or at least i couldnt reproduce a case where it broke.

The problem stems from the :: scopedInvocation. At the start of the same function, ``targetNameNode`` is overridden in case of a scoped Invocation.

So as long as it is not a scopedInvocation, ``targeteNodeName`` is set to ``targetSrc``, which is "ToScopedWChar!" in this case, the code snipped above just sets the indices to the correct ones again, no change. But if ``targetNodeName`` was overridden, it is now "mInitialDir.ToScopedNativeWChar!", so the above snippet sets the hightlight to be much longer, but of course still starts from the right location, making it select a bunch of unrelated chars after it. The '!' is removed from the selection later in every case.

**Comment by bfiete** [comment]
Fixed at b784101e93259fb9ec342c7db7f93a4e87300670
